### PR TITLE
Better separate the smart proxy installation instructions between Katello and Foreman

### PR DESCRIPTION
**Pull request by ianballou**
This PR adds some better separation between vanilla (no Katello), content, and Satellite smart proxies.

I've removed the parts that say "Katello only" and hid those sections instead.

Let me know if I missed anything.

### CODE REVIEW

**Comment by ehelms** [comment]
I'd suggest we drop `include::modules/proc_configuring-repositories-proxy.adoc[leveloffset=+1]` in favor of 
https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_configuring-repositories.adoc

File: guides/common/modules/proc_installing-capsule-server-packages.adoc
Comment (ehelms) [inline guides/common/modules/proc_installing-capsule-server-packages.adoc:16]: I'd suggest we either drop this file all together and use https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_installing-the-satellite-server-packages.adoc or create a similarly structure file for smart-proxy/capsule. As there is a lot of nuance missing here per OS.

**Comment by ehelms** [comment]
The storage requirements will need an update to account for OS versions similar to https://docs.theforeman.org/nightly/Installing_Server_on_Red_Hat/index-katello.html#storage-requirements_foreman

**Comment by ianballou** [comment]
@ehelms I updated the smart proxy storage docs.  Let me know how the numbers and columns look. I removed `/opt` from CentOS 8 to match them main install storage requirements, but there might be something I missed to add.

**Comment by ehelms** [comment]
@ianballou Looks like you need a rebase.

**Comment by lzap** [comment]
LGTM, this was rebased. Any objections to merge?

**Comment by ehelms** [comment]
When rendered, this has a section titled `2.1. Registering to Foreman server` which talks about sub-man registering the host to Foreman before installing. That is not actually required and I wonder if it should be an optional step later in the process.

File: guides/common/modules/proc_installing-capsule-server-packages.adoc
Comment (ehelms) [inline guides/common/modules/proc_installing-capsule-server-packages.adoc:37]: I think this line is breaking rendering.

**Comment by ianballou** [comment]
> When rendered, this has a section titled `2.1. Registering to Foreman server` which talks about sub-man registering the host to Foreman before installing. That is not actually required and I wonder if it should be an optional step later in the process.

@ehelms I agree that it's optional.  I feel like it's there in case the smart proxy-related packages are expected to be downloaded from synced repositories on the Katello server.  I feel like it makes more sense to have this first then to mention it later. Otherwise I'd argue to remove it completely.  What do you think about just adding a note mentioning that this step is optional?

File: guides/common/modules/proc_installing-capsule-server-packages.adoc
Comment (ianballou) [reply guides/common/modules/proc_installing-capsule-server-packages.adoc:37]: Looks like it, I've pushed the fix.

**Comment by ehelms** [comment]
> @ehelms I agree that it's optional. I feel like it's there in case the smart proxy-related packages are expected to be downloaded from synced repositories on the Katello server. I feel like it makes more sense to have this first then to mention it later. Otherwise I'd argue to remove it completely. What do you think about just adding a note mentioning that this step is optional?

I like either optional, with a note about if you plan to sync the Foreman and Katello repositories locally then you need to do this step or make this an advanced method that comes after the "default" way of using the hosted repositories directly.

**Comment by ianballou** [comment]
Alright, so I fixed it by creating a separate `common/modules/proc_registering-capsule-to-satellite-server.adoc` file since I needed to get the "optional" warning underneath the bold title.

**Comment by lzap** [comment]
@ehelms or @ekohl can we get the final blessing?

**Comment by lzap** [comment]
I am about to branch off 2.4 series, what is the status please? I can hold off the branching process if this is something that should be in 2.4/4.0. Let me know, thanks.

**Comment by ianballou** [comment]
@lzap I don't think this PR is worth holding back branching.

File: guides/common/modules/proc_installing-capsule-server-packages.adoc
Comment (ekohl) [inline guides/common/modules/proc_installing-capsule-server-packages.adoc:58]: For non-Satellite builds, can we combine it into RHEL 7 / CentOS 7 or are they really different?

File: guides/common/modules/proc_installing-capsule-server-packages.adoc
Comment (ekohl) [inline guides/common/modules/proc_installing-capsule-server-packages.adoc:50]: Don't we support RHEL 8 upstream?

**Review by ekohl — COMMENTED** [review]
I think overall this is ok and probably already an improvement. My biggest note (also inline) is that upstream I'd like to see `RHEL x / CentOS x` where the instructions are identical.

Also a note that CentOS 8 is discouraged by the CentOS project. They now prefer explicit CentOS Linux 8 or CentOS Stream 8. There may be many more of these, but I want to raise awareness to it.

File: guides/common/modules/proc_registering-capsule-to-satellite-server.adoc
Comment (ianballou) [reply guides/common/modules/proc_registering-capsule-to-satellite-server.adoc:21]: I'll remove it.

File: guides/common/modules/proc_installing-capsule-server-packages.adoc
Comment (ianballou) [reply guides/common/modules/proc_installing-capsule-server-packages.adoc:50]: I'm going to assume yes unless anyone else says otherwise.

File: guides/common/modules/ref_capsule-storage-requirements.adoc
Comment (ianballou) [reply guides/common/modules/ref_capsule-storage-requirements.adoc:24]: I'm not seeing what would need the variable. Let me know if my recent push helped at all with this.

**Review by jturel — APPROVED** [review]
LGTM

@ehelms @ekohl it appears that all feedback has been addressed for some time. is it OK to be merged?

**Comment by melcorr** [comment]
Would you mind rebasing, and I will merge! TY

**Comment by ianballou** [comment]
@melcorr I've rebased just now.

**Pull request merged**
